src/net.h

```diff
@@ -1547,6 +1577,12 @@ class CConnman
      */
     std::atomic_bool m_start_extra_block_relay_peers{false};
 
+    /**
+     * Number of `ConnectionType::PRIVATE_BROADCAST` connections to open.
+     * Whenever such a connection is opened this is decremented with 1.
+     */
+    std::atomic_size_t m_private_broadcast_connections_to_open{0};
```

Comment:
_from the main thread at https://github.com/bitcoin/bitcoin/pull/29415#pullrequestreview-2041604763_

> The main thing I'm wondering currently is why are we tacking `m_private_broadcast_connections_to_open` loosely? It feels harder to reason about, but I don't see what the benefit of it is.

The reason for this is that it is simpler to implement that way (I think, somebody has a simpler proposal?). This is because the connman thread is creating the connections, they are consumed/used by the peerman thread and the requests for new connections can come from yet another (3rd) thread. For example, while connman is creating a new connection which is needed to broadcast a given transaction, that transaction may be successfully broadcast by peerman and received back from the network, rendering the just created new connection by connman unnecessary. This is harmless and I accept it because avoiding it would make things too complicated or slow.